docs/website/docs/reference/frequently-asked-questions.md

```diff
@@ -0,0 +1,101 @@
+
+In case you don't have the source required listed in the [verified sources](../../docs/dlt-ecosystem/verified-sources/), you could create your own pipeline by refering to the [following docs](../../docs/walkthroughs/create-a-pipeline). 
+
+#### Q: How can I retrieve the complete schema of a `dlt` source?
+
+To retrieve the complete schema of a `dlt` source as `dlt` recognizes it, you need to execute both the extract and normalization steps. This process can be conducted on a small sample size to avoid processing large volumes of data. You can limit the sample size by applying an `add_limit` to your resources. Here is a sample code snippet demonstrating this process:
+
+```py
+p = dlt.pipeline(pipeline_name="my_pipeline",
+				 destination="duckdb",
+				 dataset_name="my_dataset")
+p.extract(my_source)
+p.normalize()
+print(p.default_schema.to_pretty_yaml())
+```
+
+This method ensures you obtain the full schema details, including all columns, as seen by `dlt`, without needing a predefined contract on these resources.
+
+#### Q: Can `dlt` periodically commit updates to a resource state to an external database during a pipeline run?
+
+Currently, `dlt` does not offer the functionality to checkpoint the extraction process during its execution. This limitation means that any updates to a resource state are not committed until the pipeline has fully completed its run. This issue has been recognized by the `dlt` development team and is currently being tracked under a specific GitHub issue ([#215](https://github.com/dlt-hub/dlt/issues/215)). Unfortunately, there is no specified timeline for when this feature will be available.
+
+In light of this limitation, `dlt` advises a workaround for incremental loading. This involves specifying both the start and end dates or page numbers for your data extraction process and conducting the backfill using smaller, manageable ranges, which are usually divided into around 100 loads. Implementing this strategy provides a higher degree of control and significantly mitigates the risk of losing any progress should the pipeline experience a failure or if it needs to be stopped midway through execution. For a more comprehensive understanding of how to implement this workaround, it is recommended to consult `dlt's` documentation on incremental loading with a cursor field, which can be found here: [Incremental Loading with a Cursor Field](../../docs/general-usage/incremental-loading#incremental-loading-with-a-cursor-field).
+
+#### Q: Is truncating or deleting a staging table safe?
+
+You can safely truncate those or even drop the whole staging dataset. However, it will have to be recreated on the next load and might incur extra loading time or cost.
+
+#### Q: How can I develop a "custom" pagination tracker, and are there examples available?
+
+> Can you use `dlt.sources.incremental` to create a custom cursor for tracking pagination in data streams that lack an explicit cursor field? 
```

Comment:
I see two questions here. Which one is relevant?

Reply:
corrected